### PR TITLE
refactor: Adapt to Sphinx (Refactor proof handling)

### PR DESCRIPTION
**Pull request by huitseeker**
## Summary

This is the companion PR of https://github.com/argumentcomputer/sphinx/pull/150

## In Detail

- Update Sphinx dependencies from tag v1.0.0 to branch forward_ports_43 across all components.
- Modification of all instances of `SphinxProof` in favor of `SphinxProofWithPublicValues`.
- Inclusion of the run() method call chained after the `prove` and `execute` functions across system (enacting builder pattern).
- Removal and integration of separate `prove_plonk` and `verify_plonk` functions into main proving and verification functions using the method `.plonk()` where required.
- Direct passing of `stdin` as an argument in the `execute` function across applications.
- Update of the Rust Toolchain from "nightly-2024-07-30" to "nightly-2024-08-13" in both Aptos and Ethereum components.
- Addition of `bytes` function for supporting different proof types in the fixture-generator.
- Updating proof creation and validation in the fixture generation process with revised function calls.
- Revisions in client and server operations to support `SphinxProofWithPublicValues` and updated function calls.
- Update of `tokio` version from `"1.39.0"` to `"1.39.2"` and Hex dependency addition at version `0.4.3`.

> [!IMPORTANT]
> After https://github.com/argumentcomputer/sphinx/pull/150

> [!NOTE]
> This is a companion PR, what you want to look at the CI result of the *penultimate* commit. The last commit's failure is expected.

### CODE REVIEW

File: fixture-generator/src/bin/main.rs
Comment (wwared) [inline fixture-generator/src/bin/main.rs]: @storojs72: Can you double check this part of the code? I'm not sure, but I think this might need to be `pr.encoded_proof` instead of `pr.raw_proof`. Specifically, the `encoded_proof` is returned by gnark [here](https://github.com/succinctlabs/sp1/blob/6a6c06481712eb5362dcb1dbe37a241e932dd690/crates/recursion/gnark-ffi/go/sp1/utils.go#L25) as the result of calling `MarshalSolidity`

This may also require an update to `sphinx-contracts`, not sure but we should test to make sure

**Review by wwared — COMMENTED** [review]
Everything looks good! Except I'm not 100% sure that the `bytes()` method in the fixture-generator is correct with the `pr.raw_proof`/`pr.encoded_proof` thing

Specifically, I'll try to generate some new fixtures with this updated branch to ensure those are working fine (I'm not sure but I think CI doesn't check this, since we only generate fixtures on a nightly job? I'll report back with results either way)

**Comment by huitseeker** [comment]
> Except I'm not 100% sure that the bytes() method in the fixture-generator is correct with the pr.raw_proof/pr.encoded_proof thing

True, I can revert to the encoded proof if this is needed, it's an easy fix.

**Comment by wwared** [comment]
@samuelburnham: We will need to update the `cycle-count-regression` CI check due to output format changes, see #164. It's failing for aptos in this PR and the ethereum one is currently only passing because this PR has not yet been rebased on top of #163 so it's not running any tests. This is a lower priority/non-blocking issue that we can solve later on and merge with the failing CI checks if we need to

**Comment by storojs72** [comment]
I have updated Move tests, so now only `cycle-count-regression` stuff is broken. As we can tackle this in subsequent PRs, I think this one is good to merge once CI finishes

**Review by wwared — APPROVED** [review]
Looks great!

**Review by tchataigner — CHANGES_REQUESTED** [review]
I will just test the binaries flow before we merge it

**Review by tchataigner — APPROVED** [review]
I think we are good to go. I also checked for performances and we have a nice 10% gain on proof generation.

cc @huitseeker

**Pull request merged**
